examples/prom/nonlinear_elasticity_global_rom.cpp

```diff
@@ -0,0 +1,1747 @@
+//
+// Merge phase:
+//      ./nonlinear_elasticity_global_rom --merge -ns 2 -dt 0.01 -tf 5.0
+//
+// Create FOM comparison data:
+//      ./nonlinear_elasticity_global_rom --offline -dt 0.01 -tf 5.0 -s 14 -vs 100 -sc 1.00 -id 2
+//
+// Online phase with full sampling:
+//      ./nonlinear_elasticity_global_rom --online -dt 0.01 -tf 5.0 -s 14 -vs 100 -hyp -rvdim 40 -rxdim 10 -hdim 71 -nsr 1170 -sc 1.00
+// Output message:
+//      Elapsed time for time integration loop 14.0381
+//      Relative error of ROM position (x) at t_final: 5 is 2.97882e-06
+//      Relative error of ROM velocity (v) at t_final: 5 is 0.000679751
+//
+// Online phase with strong hyper reduction:
+//      ./nonlinear_elasticity_global_rom --online -dt 0.01 -tf 5.0 -s 14 -vs 100 -hyp -rvdim 3 -rxdim 2 -hdim 4 -nsr 10 -sc 1.00
+// Output message:
+//      Elapsed time for time integration loop 1.01136
+//      Relative error of ROM position (x) at t_final: 5 is 0.0580267
+//      Relative error of ROM velocity (v) at t_final: 5 is 0.28365
+//
+// =================================================================================
+//
+// Sample runs and results for parametric ROM using only displacement basis
+// and nonlinear term basis:
+// Offline phase:
+//      ./nonlinear_elasticity_global_rom --offline -dt 0.01 -tf 5.0 -s 14 -vs 100 -sc 0.90 -xbo -def-ic -id 0
+//      ./nonlinear_elasticity_global_rom --offline -dt 0.01 -tf 5.0 -s 14 -vs 100 -sc 1.10 -xbo -def-ic -id 1
+//
+// Merge phase:
+//      ./nonlinear_elasticity_global_rom --merge -ns 2 -dt 0.01 -tf 5.0 -xbo
+//
+// Create FOM comparison data:
+//      ./nonlinear_elasticity_global_rom --offline -dt 0.01 -tf 5.0 -s 14 -vs 100 -sc 1.00 -xbo -def-ic -id 2
+//
+// Online phase with full sampling:
+//      ./nonlinear_elasticity_global_rom --online -dt 0.01 -tf 5.0 -s 14 -vs 100 -hyp -rxdim 57 -hdim 183 -nsr 1170 -sc 1.00 -xbo -def-ic
+// Output message:
+//      Elapsed time for time integration loop 18.9874
+//      Relative error of ROM position (x) at t_final: 5 is 7.08272e-05
+//      Relative error of ROM velocity (v) at t_final: 5 is 0.00387647
+//
+// Online phase with strong hyper reduction:
+//      ./nonlinear_elasticity_global_rom --online -dt 0.01 -tf 5.0 -s 14 -vs 100 -hyp -rxdim 2 -hdim 4 -nsr 10 -sc 1.00 -xbo -def-ic
+// Output message:
+//      Elapsed time for time integration loop 1.01136
+//      Relative error of ROM position (x) at t_final: 5 is 0.0130818
+//      Relative error of ROM velocity (v) at t_final: 5 is 0.979978
+
+#include "mfem.hpp"
+#include "linalg/Vector.h"
+#include "linalg/BasisGenerator.h"
+#include "linalg/BasisReader.h"
+#include "hyperreduction/DEIM.h"
+#include "hyperreduction/GNAT.h"
+#include "hyperreduction/S_OPT.h"
+#include "mfem/SampleMesh.hpp"
+
+#include <memory>
+#include <cmath>
+#include <limits>
+#include <iostream>
+#include <fstream>
+
+typedef enum {FSPACE } FESPACE;
```

Comment:
This is not used.

Reply:
You are right! I used it in the debugging phase and it stuck